mod/cornercutter/ModFeature/GlobalSetting/CoWorkingSpaceSlide.cs

```diff
@@ -0,0 +1,118 @@
+﻿using cornercutter.Enum;
+using cornercutter.ModLoader;
+using HarmonyLib;
+using System;
+using System.Collections;
+using UnityEngine;
+
+namespace cornercutter.ModFeature.GlobalSetting
+{
+    // In order to send Jackie to CWS from the slide, we are intercepting the trigger
+    // that is fired to have her exit it.
+    [HarmonyPatch(typeof(Animator), nameof(Animator.SetTrigger), new Type[] { typeof(string) })]
+    class CoWorkingSpaceSlide
+    {
+        // Flag for checking if we have been inside the slide, and
+        // so whether to run a special animation next time we go back to Fizzle
+        public static bool runAssociatedSlideAnimationNext = false;
+
+        private static bool SlideGoesToCWS()
+        {
+            // This checks, amongst the other usual ones, that we are in the gone under Fizzle,
+            // making sure you can't go to CWS unless it is actually unlocked and completed in game.
+            // The game doesn't accomodate for completing CWS (and hence go under) before
+            // you actually have the relics that let you go there in the first place
+            CutterConfig cornercutter = CutterConfig.Instance;
+            GlobalOptions globals = cornercutter.GlobalOptions;
+            return cornercutter.CornercutterIsEnabled()
+                && globals.HasFlag(GlobalOptions.SlideToCWS)
+                && SaveData.instance.goneUnder;
+        }
+
+        static bool Prefix(string name)
+        {
+            // If we are in an animation that is not the slide, run the usual code
+            if (name != "ExitSillySlide") return true;
+
+            // If we are in the slide animation AT THE MOMENT, that means we want to complete it as usual
+            // and the CWS teleport has already happened. Unset the flag and run the usual code.
+            if (runAssociatedSlideAnimationNext)
+            {
+                runAssociatedSlideAnimationNext = false;
+                return true;
+            }
+
+            if (!SlideGoesToCWS()) return true;
+
+            // Flags the slide as entered and runs the scene load code from the other in-game dungeon triggers
+            runAssociatedSlideAnimationNext = true;
+            AudioManager.PlayOneShot("Jackie_Death");
+            GUIManager.instance.LoadDungeonScene("Chaos", GlobalDungeons.dungeons.GetLoadingScreen(DungeonCompany.Chaos));
+            GUIManager.instance.tasks.HideAll();
+
+            // Don't run the usual slide exit code
+            return false;
+        }
+    }
+
+    [HarmonyPatch(typeof(EndgameScreen), "KilledBy")]
+    class RunStatus
+    {
+        public static bool? runWon = null;
```

Comment:
I'm not a fan of just having public, non-constant fields. I think this should probably be a property. I would also not make this nullish and instead make sure to set it to false instead of null.
```cs
public static bool RunWon { get; set; };

// ...

RunWon = Singleton<DungeonManager>.instance.currentRun.won ?? false;
```

Reply:
This null is being used to determine if the variable is being set at all, as this event is not guaranteed to fire. This could be spun out into two variables for clarity - for the time being though, I will make the property suggestions.